src/serial/config.rs

```diff
@@ -0,0 +1,219 @@
+use crate::prelude::*;
+use crate::time::Bps;
+
+#[derive(PartialEq, PartialOrd, Clone, Copy)]
+pub enum WordLength {
+    DataBits7,
+    DataBits8,
+    DataBits9,
+}
+
+#[derive(PartialEq, PartialOrd, Clone, Copy)]
+pub enum Parity {
+    ParityNone,
+    ParityEven,
+    ParityOdd,
+}
+
+
+#[derive(PartialEq, PartialOrd, Clone, Copy, Debug)]
+pub enum StopBits {
+    #[doc = "1 stop bit"]
+    STOP1 = 0b00,
+    #[doc = "0.5 stop bits"]
+    STOP0P5 = 0b01,
+    #[doc = "2 stop bits"]
+    STOP2 = 0b10,
+    #[doc = "1.5 stop bits"]
+    STOP1P5 = 0b11,
+}
+
+impl StopBits {
+    pub fn bits(self) -> u8 {
+        self as u8
+    }
+}
+
+#[derive(PartialEq, PartialOrd, Clone, Copy, Debug)]
+pub enum FifoThreshold {
+    #[doc = "1/8 of its depth"]
+    _1_BYTE = 0b000,
+    #[doc = "1/4 of its depth"]
+    _2_BYTES = 0b001,
+    #[doc = "1/2 of its depth"]
+    _4_BYTES = 0b010,
+    #[doc = "3/4 of its depth"]
+    _6_BYTES = 0b011,
+    #[doc = "7/8 of its depth"]
+    _7_BYTES = 0b100,
+    #[doc = "fifo empty/full"]
```

Comment:
It seems in this HAL all numeric values are as `B_12` `AF0` etc and other enums are in PascalCase.
I would change `_1_BYTE` to `FIFO_1_BYTE` or `TH_1_BYTE` or `TH2_BYTES`. Leading `_` implies that its for internal use.

Reply:
Good point. Will do